### PR TITLE
cli: {2} is reserved tenant ID, use {3} as tenant ID for demoapp

### PR DESCRIPTION
**Pull request by cthumuluru-crdb**

### CODE REVIEW

**Review by stevendanna — APPROVED** [review]
Thanks! You'll almost certainly need to update some tests to account for this change.

Would you mind filing an issue to remind us that we should consider updating the minimum tenant ID constant?  I'm not sure that will actually be possible. But it would be good to have a placeholder to remind us to decide that.

**Review by cthumuluru-crdb — COMMENTED** [review]
> You'll almost certainly need to update some tests to account for this change.

Sure. I'll take a look at the tests to update. 

> Would you mind filing an issue to remind us that we should consider updating the minimum tenant ID constant? I'm 
not sure that will actually be possible. But it would be good to have a placeholder to remind us to decide that.

I already have a commit in my branch with that update. I'll definitely create a ticket to capture the discussion on making that change. 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/131805#-:-O8IMRle0RfPIxX_ya54:b-wcdll0)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @shubhamdhama)

File: pkg/cli/democluster/demo_cluster_test.go [hunk 1]
```diff
@@ -406,7 +406,7 @@ func TestTenantCapabilities(t *testing.T) {
 		if cap == tenantcapabilities.TenantSpanConfigBounds {
 			capValue = `{}`
 		}
-		expectedRows = append(expectedRows, []string{`2`, demoTenantName, `ready`, `shared`, cap.String(), capValue})
+		expectedRows = append(expectedRows, []string{`3`, demoTenantName, `ready`, `shared`, cap.String(), capValue})
```
Comment (stevendanna) [inline pkg/cli/democluster/demo_cluster_test.go:409]: :D It is a rare day when we change something like this and a single test fails.

**Review by stevendanna — APPROVED** [review]

File: pkg/cli/democluster/demo_cluster_test.go (hunk 1 above)
Comment (cthumuluru-crdb) [reply pkg/cli/democluster/demo_cluster_test.go:409]: I got lucky then.. :)

**Pull request merged**